[IMP] Manufacturing: Resolving work order dependency problems (#17301)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/advanced_configuration/work_order_dependencies.rst

Reviewer: Suggestion: Adding an image here would enhance the example.

.. |BOM| replace:: :abbr:`BoM (Bill of Materials)`

When manufacturing certain products, specific operations may need to be completed before others can

> begin. To ensure operations are carried out in the correct order, the *Manufacturing* app features a

Reviewer: ```suggestion
begin. To ensure operations are carried out in the correct order, the **Manufacturing** app features a
```
Correction: App names should be bold.

*work order dependencies* setting. Enabling this setting allows for operations on a Bill of

> Materials (BoM) to be *blocked* by other operations that should occur first.

Reviewer: ```suggestion
*work order dependencies* setting. Enabling this setting blocks the Bill of
Materials (BoM) operations when other operations should occur first.
```
Suggestion: revise to make it direct and concise.

Configuration
=============

The *work order dependencies* setting is not enabled by default. To enable it, begin by navigating

> to :menuselection:`Manufacturing app --> Configuration --> Settings`. Then, enable the

Reviewer: ```suggestion
to the :menuselection:`Manufacturing app --> Configuration --> Settings`. Then, enable the
```
Correction: added 'the' before the menu selection.

   :alt: The Operation Dependencies checkbox on the Miscellaneous tab of a BoM.

Open the *Operations* tab. Click the :icon:`oi-settings-adjust` :guilabel:`(settings)` icon, then
enable the :guilabel:`Blocked By` checkbox. The :guilabel:`Blocked By` field appears for each
operation in the tab.

.. image:: work_order_dependencies/operations-settings.png
   :alt: The settings for the Operations tab on a BoM.

In the line of the operation that should be blocked by another operation, click the
:guilabel:`Blocked By` field. The *Open: Operations* pop-up window opens. In the :guilabel:`Blocked
By` field, select the blocking operation that must be completed *before* the operation that is
blocked.

.. image:: work_order_dependencies/blocked-by.png
   :alt: The Blocked By drop-down field for an operation on a BoM.

Finally, save the |BOM| by clicking :icon:`fa-cloud-upload` :guilabel:`(Save manually)`.

.. seealso::
   :doc:`../basic_setup/bill_configuration`

Plan work orders using dependencies
===================================

After work order dependencies are configured for a |BOM|, the *Manufacturing* app can plan when work

> orders are scheduled based on those dependencies. To plan the work orders for a manufacturing order,

Reviewer: ```suggestion
After work order dependencies are configured for a |BOM|, the *Manufacturing* app can plan work
order schedules based on those dependencies. To plan the work orders for a manufacturing order,
```
Suggestion: shortened the sentence.

navigate to :menuselection:`Manufacturing app --> Operations --> Manufacturing Orders`.

Next, select a manufacturing order for a product with work order dependencies set on its |BOM|, or
create a new manufacturing order by clicking :guilabel:`New`. If a new manufacturing order is
created, select a |BOM| configured with work order dependencies from the :guilabel:`Bill of
Material` drop-down field, then click :guilabel:`Confirm`.

> After confirming the manufacturing order, open the *Work Orders* tab to view the work orders

Reviewer: ```suggestion
After confirming the manufacturing order, open the :guilabel:`Work Orders` tab to view the work orders
```
Correction: If the tab is part of the instruction action, then it should be a guilabel.

Planning by workcenter
----------------------

To see a visual representation of how work orders are planned, open the *Work Orders Planning* page
by navigating to :menuselection:`Manufacturing app --> Planning --> Planning by Workcenter`. This
page shows a timeline of all work orders scheduled for each operation.

If a work order is blocked by the completion of another, the blocked work order is shown as

> scheduled to start after the work order that blocks it. In addition, an arrow connects the two work

Reviewer: ```suggestion
If a work order is blocked by another, the blocked work order is shown as
scheduled to start after the blocking work order. In addition, an arrow connects the two work
```
Suggestion: Revision to make this sentence less wordy and improve clarity.

orders, leading from the blocking operation to the blocked operation.

.. image:: work_order_dependencies/planning-arrow.png
   :alt: The arrow connecting a blocked work order to the work order blocking it.

Resolving inconsistencies
~~~~~~~~~~~~~~~~~~~~~~~~~

The arrows on the *Work Orders Planning* page can also point out inconsistencies in the
manufacturing order and how to correct them.

.. example::
   A workshop is manufacturing a table using three work orders: Assemble, then apply finish, and

>    finally, dry.

Reviewer: Are the work orders supposed to be capitalized?

   If the second work order (apply finish) is moved before the first work order, a red line appears

>    linking the two work orders. The line is red to indicate that the moved work order violates the

Reviewer: ```suggestion
   linking the two. The line is red to indicate that the moved work order violates the
```
Suggestion

   defined work order dependencies.

   .. image:: work_order_dependencies/out-of-order.png
      :alt: A red line appears between two work orders.

Using icons on the line
***********************

When hovering over the connecting line, three icons appear. The left and right arrows can be used to
reschedule work orders to help maintain the correct work order dependencies.

.. image:: work_order_dependencies/three-icons.png
   :alt: A left arrow, X, and right arrow appear on the line for out-of-order work orders.

- Left arrow: The work order is scheduled to be performed earlier than its linked step.
- Cancel button: Odoo assumes the dependencies no longer apply due to an exception, and the link
  between the steps is broken.

> - Right arrow: The work order is scheduled to be performed later than its linked step.

Reviewer: ```suggestion
- **Left arrow**: The work order is scheduled to be performed earlier than its linked step.
- **Cancel button**: Odoo assumes the dependencies no longer apply due to an exception, and the link
  between the steps is broken.
- **Right arrow**: The work order is scheduled to be performed later than its linked step.
```
Suggestion: If there is no specific icon guilabel, then bold the button options.

In the manufacturing order
**************************

Problems with scheduled work orders are visible in the *Work Orders* tab of the manufacturing order.
Problems are indicated by the red :icon:`fa-exclamation-triangle` :guilabel:`(exclamation triangle)`
icon on the work order line.

> .. image:: work_order_dependencies/mo-scheduling-problem.png

Reviewer: Suggestion: I know it would be a hassle, but I recommend retaking screenshots that include the columns and ensuring the column titles are fully visible. This ensures there is enough context for the reader.

   :alt: A scheduling problem shows in the second operation in a group of three work orders.

Click the red :icon:`fa-exclamation-triangle` :guilabel:`(exclamation triangle)` icon, then click
:guilabel:`Replan` in the *Scheduling Information* pop-up window. The work order is rescheduled to

> maintain the work order dependencies.

Reviewer: ```suggestion
maintain its dependencies.
```
Suggestion
